File: MobilSmartWear/app/src/main/java/com/example/mobilsmartwear/data/model/Product.kt
package com.example.mobilsmartwear.data.model

import com.example.mobilsmartwear.data.remote.RetrofitClient
import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("_id")
    val id: String,
    val name: String,
    val description: String,
    val price: Double,
    @SerializedName("image")
    private val _imageUrl: String,
    val category: String,
    val sizes: List<String> = listOf("S", "M", "L", "XL"),
    @SerializedName("stock")
    val stockCount: Int = 100,
    val subcategory: String? = null,
    val brand: String,
    @SerializedName("featured")
    val featured: Boolean = false,
    @SerializedName("salesCount")
    val salesCount: Int = 0,
    val color: String? = null,
    val size: String? = null,
    @SerializedName("createdAt")
    val createdAt: String? = null,
    var isFavorite: Boolean = false,
    @SerializedName("rating")
    val rating: Float = 0f,
    val tags: List<String>? = null,
    @SerializedName("combinationCode")
    val combinationCode: String? = null
) {
    val imageUrl: String
        get() = RetrofitClient.getImageUrl(_imageUrl)

    fun hasStock(): Boolean {
        return stockCount > 0
    }
} 

File: WEBSmartWear/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    description: {
        type: String,
        required: true
    },
    price: {
        type: 'Double', // MongoDB Double veri tipi
        required: true,
        min: 0
    },
    category: {
        type: String,
        required: true,
        enum: ['Giyim', 'Ayakkabı', 'Aksesuar']
    },
    subcategory: {
        type: String,
        default: ''
    },
    image: {
        type: String,
        required: true
    },
    stock: {
        type: Number,
        required: true,
        min: 0
    },
    featured: {
        type: Boolean,
        default: false
    },
    color: {
        type: String,
        default: ''
    },
    size: {
        type: String,
        default: '',
        validate: {
            validator: function(size) {
                // Giyim kategorisi için zorunlu
                if (this.category === 'Giyim' && (!size || size === '')) {
                    return false;
                }
                return true;
            },
            message: 'Giyim ürünleri için beden bilgisi gereklidir.'
        }
    },
    brand: {
        type: String,
        default: 'SmartWear'
    },
    combinationCode: {
        type: String,
        enum: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13'],
        required: true,
        description: {
            '1': 'Ayakkabı',
            '2': 'Pantolon',
            '3': 'Kemer',
            '4': 'Basic Tişört',
            '5': 'Basic Üstüne Giyilebilir',
            '6': 'Tekli Üst Giyim',
            '7': 'Ceket/Mont',
            '8': 'Atkı',
            '9': 'Şapka',
            '10': 'Saat',
            '11': 'Takım',
            '12': 'Cüzdan',
            '13': 'Eldiven'
        }
    },
    createdAt: {
        type: Date,
        default: Date.now
    }
}, {
    toJSON: { getters: true },
    toObject: { getters: true }
});

// Virtuals
productSchema.virtual('availableSizes').get(function() {
    return this.size ? this.size.split(',') : [];
});

// Middleware
productSchema.pre('save', function(next) {
    // Giyim kategorisi için varsayılan bedenler
    if (this.category === 'Giyim' && (!this.size || this.size === '')) {
        this.size = 'XS,S,M,L,XL';
    }
    next();
});

module.exports = mongoose.model('Product', productSchema);
